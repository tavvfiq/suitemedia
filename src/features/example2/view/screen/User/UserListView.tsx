import React from 'react';
import { ScrollView, View, Text, StyleSheet } from 'react-native';
import Overlay from '~core/view/overlay/Overlay';
import User from '~features/example2/data/domain/user';
import ArticlePlaceholder from '../../component/ArticlePlaceholder';

interface Props {
  users: User[];
  error?: boolean;
  loading?: boolean;
}

const UserListView = ({ users, error, loading }: Props) => {
  return (
    <ScrollView contentContainerStyle={styles.container}>
      {loading && <ArticlePlaceholder />}
      {users.map(user => (
        <Text style={styles.text} key={user.id}>
          {user.name}
        </Text>
      ))}
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    color: 'blue',
  },
  text: {
    color: 'black',
  },
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  button: {
    backgroundColor: 'red',
    flexGrow: 1,
    height: 100,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default UserListView;
