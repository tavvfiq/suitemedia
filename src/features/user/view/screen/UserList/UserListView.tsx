import { FlatList, ListRenderItem, View } from 'react-native';
import React from 'react';
import Header from '~common/view/component/Header';
import MapIcon from '~features/user/assets/svg/MarkerIcon';
import Menu from '~features/user/assets/svg/Menu';
import User, { UserI } from '~features/user/data/model/user';
import Placeholder from '../../component/List/Placeholder';
import List from '../../component/List';
import UserMapView from '../../component/UserMapView';
import { generateRandomLat, generateRandomLong } from '~core/utils/generator';

type Props = {
  users?: User[];
  onBackPress: () => void;
  loading: boolean;
  userOnPress: (user: UserI) => void;
  markerOnPress: (userId?: string) => void;
  onRefresh: () => void;
  onLoadMore: () => void;
  isRefreshing: boolean;
};

const UserListView = ({
  users = [],
  onBackPress,
  userOnPress,
  markerOnPress,
  onRefresh,
  onLoadMore,
  isRefreshing,
  loading,
}: Props) => {
  const [isMapView, toggleMapView] = React.useState(false);
  const rightIconOnPress = function () {
    toggleMapView(!isMapView);
  };

  const renderRightIcon = function () {
    return isMapView ? <Menu /> : <MapIcon />;
  };

  const renderUsers: ListRenderItem<User> = ({ item }) => {
    return <List user={item} onPress={userOnPress} />;
  };

  const renderEmptyUsers = () => {
    if (loading) {
      return <Placeholder />;
    }
    return <View />;
  };

  const renderUserList = () => {
    return !isMapView ? (
      <FlatList
        data={users}
        renderItem={renderUsers}
        keyExtractor={item => item.id.toString()}
        ListEmptyComponent={renderEmptyUsers}
        ListFooterComponent={renderEmptyUsers}
        refreshing={isRefreshing}
        onRefresh={onRefresh}
        onEndReached={onLoadMore}
        onEndReachedThreshold={0.1}
      />
    ) : (
      <UserMapView
        initLat={parseFloat(generateRandomLat())}
        initLng={parseFloat(generateRandomLong())}
        users={users}
        markerOnPress={markerOnPress}
      />
    );
  };

  return (
    <>
      <Header
        title="Users"
        leftIconOnPress={onBackPress}
        rightIcon={renderRightIcon()}
        rightIconOnPress={rightIconOnPress}
      />
      {renderUserList()}
    </>
  );
};

export default UserListView;
