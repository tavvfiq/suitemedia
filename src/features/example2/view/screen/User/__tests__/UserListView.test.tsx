import React from 'react';
import { render } from '@testing-library/react-native';
import User from '~features/example2/data/domain/user';
import UserListView from '../UserListView';

test('UserListView renders correctly', () => {
  const users: User[] = [];
  const loading = false;
  const error = false;
  const view = render(<UserListView users={users} loading={loading} error={error}/>).toJSON();
  expect(view).toMatchSnapshot();
})
