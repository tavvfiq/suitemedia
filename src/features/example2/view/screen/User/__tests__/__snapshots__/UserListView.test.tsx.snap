// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListView renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "color": "blue",
      "flexGrow": 1,
    }
  }
>
  <View />
</RCTScrollView>
`;
